### PR TITLE
Encryption

### PR DESCRIPTION
**Pull request by KaiNissen**
This adds an encryption module to the banner library. As we already discussed, this should all be moved into the fundraising repository. Since there is some ongoing work based on the banner lib, I will move this once all of it has been merged.

### CODE REVIEW

File: fundraising/banner.config.js [hunk 1]
```diff
@@ -8,6 +8,11 @@
 	'use strict';
 	
 	Banner.config = {
+		encryption: {
+			libUrl: '../fundraising/openpgp.min.js',
+			publicKey: '',
+			elementId: ''
```
Comment (wmde-manicki) [inline fundraising/banner.config.js]: I am wondering how the `elementId` is intended to be used. I have seen it being used in the test but it still not clear why it should be put in the config. The jquery element getting encrypted text is passed to the `encrypt` function so this is not intended for this, is it?

File: fundraising/banner.encryption.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	/**
+	 * Load the encryption library
+	 */
+	EP.initCryptLib = function () {
+		$.ajax( {
+			url: Banner.config.encryption.libUrl,
+			dataType: 'script',
+			cache: true,
+			success: function() {
+				console.log( 'encryption library loaded' );
```
Comment (wmde-manicki) [inline fundraising/banner.encryption.js]: debug message?

**Comment by wmde-manicki** [comment]
Apart from comments above looks good!

I think the Banner-JS stuff could be moved over to the other place once this PR is merged. I was thinking of adding some more stuff related to CI etc, so I think it makes sense to move as soon as possible to reduce the number of things to move.

**Comment by KaiNissen** [comment]
While testing this, I noticed that we might need to be careful, how much data we pass. The form data is being encrypted and base64 encoded, so we have quite some overhead. Especially with 2048 bit keys, we might hit that limit. We could use 1024 bit keys, but those will be factored soon, I guess.

**Comment by wmde-manicki** [comment]
Good point. I think this should be taken into account in the banner-api communication (sending data in batches of allowed length etc). I will have this in mind while working on the stuff I am currently writing.

**Pull request merged**